README.md

```diff
@@ -17,83 +17,318 @@ Main characteristics:
+
+- **Performance:**
+  - `computed`: Better performance by re-running the function only based on changes in the stores involved in the last run.
+  - `derived`: Re-run the function each time a dependent store changes.
+
+- **Asynchronous State:**
+  - `computed`: Unable to manage asynchronous state.
+  - `derived`: Can handle asynchronous state with the `set` method.
+
+- **Skipping Value Emission:**
+  - `computed`: Does not provide a mechanism to skip emitting values.
+  - `derived`: Allows skipping the emission of values by choosing not to call the provided `set` method.
+
+- **Setup and Teardown:**
+  - `computed`: Lacks explicit setup and teardown methods.
+  - `derived`: Supports setup and teardown methods, allowing actions such as adding or removing DOM listeners.
+
+While `computed` feels more intuitive in many use-cases, `derived` excels in scenarios where `computed` falls short, particularly in managing asynchronous state and providing more granular control over value emissions.
+
+Carefully choosing between them based on specific requirements enhances the effectiveness of state management in your application.
+
+### Getting the value
+
+There are three ways for getting the value of a store:
+
+```typescript
+import {writable, get} from "@amadeus-it-group/tansu";
+
+const count$ = writable(1);
+const unsubscribe = count$.subscribe((count) => {
+  // Will be called with the updated value, synchronously first, then each time count$ changes
+  // must be unsubscribe to end the future calls.
+  console.log(count);
+});
+
+// A store is also a function that you can call to get the instant value.
+console.log(count$());
+
+// Equivalent to
+console.log(get(count$));
+```
+
+> [!NOTE]
+> Getting the instant value implies the subscription and unsubription on the store:
+>   - It can be important to know in case of setup/teardown functions.
+>   - In the same scope, prefer to store the value once in a local variable instead of calling `store$()` several times.
+>
+> `get` can work with Class variant of a store,
+> When called inside a reactive context (i.e. inside a computed), getting the value serves to know and "listen" the dependent stores.
+
+
+### batch
+
+[api documentation](https://amadeusitgroup.github.io/tansu/functions/batch.html)
+
+Contrary to other libraries like Angular with signals or Svelte with runes, where the callback of a subscription is executed asynchronously (with `useEffect`), we have maintained the constraint of synchronicity between the store changes and their subscriptions in Tansu.
+
+While it is acceptable for these frameworks to defer these calls since their goals are well-known in advance (to optimize their final rendering), this is not the case for Tansu, where the goal is to be adaptable to any situation.
+
+The problem with synchronous subscription is that it can create "glitches". Subscribers and computed store callbacks that are run too many times can create incorrect intermediate values.
```

Comment:
```suggestion
The problem with synchronous subscriptions is that it can create "glitches". Subscribers and computed store callbacks that are run too many times can create incorrect intermediate values.
```